suite-common/wallet-graph/src/graphThunks.ts

```diff
@@ -5,146 +5,333 @@ import { getBlockbookSafeTime } from '@suite-common/suite-utils';
 import TrezorConnect from '@trezor/connect';
 import { Account } from '@suite-common/wallet-types';
 import { getFiatRatesForTimestamps } from '@suite-common/fiat-services';
-import { selectAccountsByNetworkSymbols } from '@suite-common/wallet-core';
+import { selectAccountsByNetworkSymbols, selectAccountByKey } from '@suite-common/wallet-core';
+import { FiatCurrencyCode } from '@suite-common/suite-config';
 
-import { LineGraphTimeFrameValues } from './types';
-import { getLineGraphAllTimeStepInMinutes } from './graphUtils';
+import {
+    LineGraphTimeFrameItemAccountBalance,
+    LineGraphTimeFrameValues,
+    LineGraphTimeFrameConfiguration,
+    GraphSection,
+} from './types';
+import {
+    enhanceBlockchainAccountHistory,
+    ensureHistoryRates,
+    getSuccessAccountBalanceMovements,
+    getTimeFrameConfiguration,
+    sortTimeFrameItemsByTimeAsc,
+} from './graphUtils';
 import { timeSwitchItems } from './config';
 import { actionPrefix } from './graphActions';
 
-const getOldestAccountBalanceMovementTimestamp = async (accounts: Account[]) => {
-    const promises = accounts.map(async account => {
-        const response = await TrezorConnect.blockchainGetAccountBalanceHistory({
-            coin: account.symbol,
-            descriptor: account.descriptor,
-            groupBy: 3600 * 24, // day
-        });
-        try {
-            if (response?.success) {
-                const { payload } = response;
-                return payload.map(balanceGroup => balanceGroup.time);
-            }
-        } catch (error) {
-            // eslint-disable-next-line no-console
-            console.log(error);
-        }
-        return [];
+type GetGraphPointsForAccountsThunkPayload = {
+    section: GraphSection;
+    fiatCurrency: FiatCurrencyCode;
+    timeFrame: LineGraphTimeFrameValues;
+};
+
+type GetGraphPointsForSingleAccountThunk = {
+    accountKey: string;
+    fiatCurrency: FiatCurrencyCode;
+    timeFrame: LineGraphTimeFrameValues;
+};
+
+const fetchAccountBalanceHistory = async (
+    account: Account,
+    { from, to, groupBy }: { from?: number; to?: number; groupBy: number },
```

Comment:
I would probably call `groupByInMs` because just number type is cryptic. I recommend you to use these suffix everytime when dealing with some time.